Library/Homebrew/utils.rb

```diff
@@ -770,3 +770,24 @@ def number_readable(number)
   (numstr.size - 3).step(1, -3) { |i| numstr.insert(i, ",") }
   numstr
 end
+
+def truncate_text_to_approximate_size(s, max_bytes, options = {})
+  front_weight = options[:front_weight] || 0.5
+  glue = options[:glue] || "\n[...snip...]\n"
+  return s if s.bytesize <= max_bytes
+
+  if front_weight < 0 || front_weight > 1
+    raise "opts[:front_weight] must be between 0.0 and 1.0"
+  end
+  scale = 0.99 # just a tad small to allow for the glue
+  while true
```

Comment:
I don't think that's the case here, because UTF-8 is a variable-length encoding. You may be able to express it more succinctly, but you have to traverse the string from the beginning, and traverse the string from the end _with a look-behind window_ to go the other way. That's the downside of UTF-8: you do not have O(1) random access to array elements when viewing it as a character array.

The loop here is a way to do "chunky" truncation without paying for the more expensive iteration over variable-length Unicode code units.